### PR TITLE
[swift4] Use RequestBuilder directly for RxSwift and PromiseKit

### PR DESCRIPTION
**Pull request by d-date**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

When we specify `RxSwift` or `PromiseKit` to `ResponseAs` in `additional_properties`, the generator will generate 3 methods, requestBuilder, default implementation and wrapped method.

We need only requestBuild and wrapped method so that reduce generated codes.
This is not breaking change for users.

Before:
```swift
open class func addPet(pet: Pet) -> Observable<Void> {
        return Observable.create { observer -> Disposable in
            addPet(pet: pet) { data, error in
                if let error = error {
                    observer.on(.error(error))
                } else {
                    observer.on(.next(data!))
                }
                observer.on(.completed)
            }
            return Disposables.create()
        }
    }
```

After:
```swift
open class func addPet(pet: Pet) -> Observable<Void> {
        return Observable.create { observer -> Disposable in
            addPetWithRequestBuilder(pet: pet).execute { (response, error) -> Void in
                if let error = error {
                    observer.onError(error)
                } else {
                    observer.onNext(())
                }
                observer.onCompleted()
            }
            return Disposables.create()
        }
    }
```

### CODE REVIEW

**Comment by d-date** [comment]
### Committes
@wing328  @jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @d-date (2018/03)

File: modules/openapi-generator/src/main/resources/swift4/api.mustache [hunk 1]
```diff
@@ -54,7 +55,7 @@ open class {{classname}} {
         {{/returnType}}
         }
     }
-
+{{/useRxSwift}}{{/usePromiseKit}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/swift4/api.mustache]: @d-date minor suggestion: using the following instead to skip blank lines or trailing spaces in the output:
```
    {{^usePromiseKit}}
    {{^useRxSwift}}
....
    {{/useRxSwift}}
    {{/usePromiseKit}}
```

Comment (d-date) [reply modules/openapi-generator/src/main/resources/swift4/api.mustache]: @wing328  Thank you! Fix it.

**Comment by wing328** [comment]
The Travis CI issue below has been fixed in master:
```
Fetching gem metadata from https://rubygems.org/.........
ZenTest-4.11.1 requires rubygems version < 3.0, which is incompatible with the
current version, 3.0.0
```

**Pull request merged**